torch/optim/sgd.py

```diff
@@ -161,6 +165,8 @@ def step(self, closure=None):
                 state = self.state[p]
                 state['momentum_buffer'] = momentum_buffer
 
+        if not self.defaults['differentiable']:
```

Comment:
nit: it would be better to have this in a try: finally: bloc
Maybe you can have a small context manager specialized for this that you apply to the step function and that we can re-use for all optimizers?

Reply:
done!